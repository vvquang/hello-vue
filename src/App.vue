<template>
  <div id="app" class="container">
    <Header />
    <router-view />
  </div>
</template>

<script>
import Header from "./pages/Header";

export default {
  name: "App",
  components: {
    Header
  }
};
</script>

<style>
* {
  box-sizing: border-box;
}

.container {
  width: 100%;
  height: 100%;
}

#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  /* text-align: center; */
  /* color: #2c3e50; */
  /* margin-top: 60px; */
}
</style>
