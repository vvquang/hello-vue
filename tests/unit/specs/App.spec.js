import { shallowMount } from '@vue/test-utils'
import App from '@/App'
import wrapperOps from '../supports/wrapper'

const wrapper = shallowMount(App, wrapperOps)

describe('test App.vue', () => {
  it('render app', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
