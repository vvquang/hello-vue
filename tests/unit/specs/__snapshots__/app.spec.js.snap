// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App.vue render app 1`] = `
<div id="app" class="container">
  <header class="header">
    <ul class="header__content">
      <li>
        <router-link-stub to="[object Object]" tag="a" event="click">Home</router-link-stub>
      </li>
      <li>
        <router-link-stub to="[object Object]" tag="a" event="click">Profile</router-link-stub>
      </li>
      <li>
        <router-link-stub to="[object Object]" tag="a" event="click">About</router-link-stub>
      </li>
    </ul>
  </header>
  <main class="container__content">
    <router-view-stub name="default"></router-view-stub>
  </main>
  <footer class="footer">
    <div class="footer__content">
      <ul class="footer__content--items">
        <li><a href="/help">Help </a></li>
        <li><a href="/about">About</a></li>
      </ul> <span class="footer__content--copyright">Â© 2020 1PAC Vietnam Co.,Ltd</span>
    </div>
  </footer>
</div>
`;
